### ISSUE TITLE
rewire_directed_edges should not create multi-edges

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1527]
`igraph_rewire_directed_edges` may create multi-edges in the `IN` and `OUT` modes, but not in the `ALL` mode.

This is inconsistent. Either it should not create multi-edges, or there should be an argument which controls whether it is allowed to create multi-edges (just like `igraph_rewire_edges`).

**Comment by szhorvat** [comment]
I have reopened this because the real task is to implement avoiding multi-edges for the directed case, and make that controllable by a parameter. This will not go into 0.9.x because (1) it will be API-breaking (2) it is not-trivial to implement.

**Comment by ntamas** [comment]
Exactly, thanks. The issue was closed automatically because of the commit that referenced the issue.
